rainier-core/src/main/scala/com/stripe/rainier/core/RandomVariable.scala

```diff
@@ -54,29 +54,29 @@ class RandomVariable[+T](private val value: T,
       .map(_.head.apply())
       .toList
 
-  def checkConvergence[V](sampler: Sampler,
-                          warmupIterations: Int,
-                          iterations: Int,
-                          chains: Int)(
+  def sampleWithDiagnostics[V](sampler: Sampler,
+                               warmupIterations: Int,
+                               iterations: Int,
+                               chains: Int)(
       implicit rng: RNG,
-      sampleable: Sampleable[T, V]): (List[V], List[Double]) = {
+      sampleable: Sampleable[T, V]): (List[V], List[Diagnostics]) = {
```

Comment:
I'd encourage return types on all public methods. If you run the `scalafix` command it will attempt to add them for you and then you can clean them up by hand.

Reply:
agreed, but where was this comment meant for? this method has a return type

Reply:
Apparently I was having a hard time reading when I wrote this original comment. (Maybe writing too much Java/Bash/Go/Python lately has ruined me for Scala.)